### PR TITLE
Feature: `VariableMenu` component

### PR DESCRIPTION
**Pull request by znicholasbrown**
Closes: #1289

### CODE REVIEW

File: src/components/VariableMenu.vue [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  const deleteVariable = async (id: string): Promise<void> => {
+    closeDeleteModal()
+
+    try {
+      await api.variables.deleteVariable(id)
+      showToast(localization.success.delete(localization.info.variable), 'success')
+    } catch (error) {
+      console.error(error)
+      showToast(localization.error.delete(localization.info.variable.toLowerCase()), 'error')
+      emit('delete', id)
```
Comment (zhen0) [inline src/components/VariableMenu.vue:60]: Is the emit meant to be in the catch block?

File: src/components/VariableMenu.vue [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const { open: openEditModal } = useShowModal()
+
+  const api = useWorkspaceApi()
+
+  const deleteVariable = async (id: string): Promise<void> => {
+    closeDeleteModal()
+
+    try {
+      await api.variables.deleteVariable(id)
+      showToast(localization.success.delete(localization.info.variable), 'success')
```
Comment (zhen0) [inline src/components/VariableMenu.vue:56]: Should the success toast also be lower case as the error one is?

File: src/components/VariableMenu.vue (hunk 1 above)
Comment (znicholasbrown) [reply src/components/VariableMenu.vue]: Nope, great catch!

File: src/components/VariableMenu.vue (hunk 2 above)
Comment (znicholasbrown) [reply src/components/VariableMenu.vue:56]: Nah - this one places the passed string at the start of the sentence; the error one places it at the end; good looks though, maybe the other method should be making things lowercase by default? 🤔  Won't tackle that in this PR though

**Review by pleek91 — APPROVED** [review]

**Pull request merged**